python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -344,6 +344,38 @@ def groups(self):
             zip(group_names.to_pandas(), grouped_index._split(offsets[1:-1]))
         )
 
+    @cached_property
+    def indices(self):
+        """
+        Dict {group name -> group indices}.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> data = [[10, 20, 30], [10, 30, 40], [40, 50, 30]]
+        >>> df = cudf.DataFrame(data, columns=["a", "b", "c"])
+        >>> df
+            a   b   c
+        0  10  20  30
+        1  10  30  40
+        2  40  50  30
+        >>> df.groupby(by=["a"]).indices
+        {10: array([0, 1]), 40: array([2])}
+        """
+        group_names, offsets, _, grouped_values = self._grouped()
+
+        if len(group_names) > self._MAX_GROUPS_BEFORE_WARN:
+            warnings.warn(
+                f"GroupBy.indices performance scales poorly with "
+                f"number of groups. Got {len(group_names)} groups."
+            )
```

Comment:
**question**: Is this warning now really still true? previously, doing `grouped_values._split(offsets[1:-1])` took _aaaages_, now with `np.split(...)` it takes a few milliseconds for 2000 groups for me.

Reply:
Yup, removed it.